packages/block-library/src/navigation/index.php

```diff
@@ -117,7 +104,23 @@ private static function is_interactive( $attributes, $inner_blocks ) {
 	 * @return bool Returns whether or not a block needs a list item wrapper.
 	 */
 	private static function does_block_need_a_list_item_wrapper( $block ) {
-		return in_array( $block->name, static::$needs_list_item_wrapper, true );
+
+		/**
+		 * Filter the list of blocks that need a list item wrapper.
+		 *
+		 * Affords ability to customize which blocks need a list item wrapper when rendered 
+		 * within a core/navigation block.
+		 * This is useful for blocks that are not list items but should be wrapped in a list
+		 * item when used as a child of a navigation block.
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param array $needs_list_item_wrapper The list of blocks that need a list item wrapper.
+		 * @return array The list of blocks that need a list item wrapper.
+		 */
+		$needs_list_item_wrapper = apply_filters( 'block_core_navigation_needs_list_item_wrapper', static::$needs_list_item_wrapper );
```

Comment:
I'm still not 100% convinced on the naming of the filter (even though I came up with it 😅 ).

`block_core_navigation_blocks_requiring_list_item_wrapper`?

Otherwise it might seem from the filter name that we are filtering the actual Navigatino block, rather than a list of blocks that require an `<li>` wrapper.

Reply:
Agree this is more clear :)